openedx/openedx-platform
# [BD-13] Deprecate ModuleSystem.render_template

## Description

Builds on the shim added by https://github.com/edx/edx-platform/pull/28440 to deprecate the `ModuleSystem.render_template` attribute, in favor of using the added `MakoService.render_template`.

This change is only a refactoring, and should not affect Learners, Course Authors, or anyone else using edx-platform.

## Supporting information

* [Content Core Platform Simplication project](https://openedx.atlassian.net/wiki/spaces/AC/pages/3091629503/Content+Core+Platform+Simplification)
* [BLENDED-113](https://openedx.atlassian.net/browse/BLENDED-113) (edX internal ticket)
* [FAL-2304](https://tasks.opencraft.com/browse/FAL-2304) (OpenCraft internal ticket)

## Testing instructions

* LMS: https://pr28571.sandbox.opencraft.hosting/
* Studio: https://studio.pr28571.sandbox.opencraft.hosting/

LMS as learner:
1. Login as a demo learner (`honor@example.com` / `edx`)
2. Browse around the [BD-13 test course](https://discovery.pr28571.sandbox.opencraft.hosting/learning/course/course-v1:OpenCraft+BD-13+2021_1/home) and ensure that blocks render as expected, and that you can submit answers to problems. This course has all of the affected XBlocks in it, including an example Video block which shows the CC license wrapper (cf 7fd14678c2f3b4f772e3ae27b254082a34bfee7e).
3. Switch to the Legacy Course Experience and browse around in that too.

LMS as course staff:
1. Login as a demo staff user (`staff@example.com` / `edx`)
2. Navigate to a course and its Instructor Dashboard > Email tab
3. Ensure that you can see and use the HTML WYSIWYG editor to craft emails to students. (cf 96d92083748145aa41001d85a28f293b338239ff)

Studio:
1. Login as a demo staff (`staff@example.com` / `edx`)
2. Navigate to the [BD-13 test course](https://studio.pr28571.sandbox.opencraft.hosting/course/course-v1:OpenCraft+BD-13+2021_1) and browse through the blocks. Ensure they render and edit as expected.
3. Check that you can submit answers to problems, and save changes to blocks.
4. Preview unpublished changes and ensure they render as expected.
4. Publish changes, and ensure they render in the LMS as expected.

## Deadline

None
